pkg/trace/api/api.go

```diff
@@ -354,6 +375,43 @@ func (r *HTTPReceiver) logStats() {
 	}
 }
 
+func (r *HTTPReceiver) watchdog(now time.Time) {
+	wi := watchdog.Info{
+		Mem: watchdog.Mem(),
+		CPU: watchdog.CPU(now),
+	}
+
+	rate, err := sampler.CalcPreSampleRate(r.conf.MaxCPU, wi.CPU.UserAvg, r.PreSampler.RealRate())
+	if err != nil {
+		log.Warnf("problem computing pre-sample rate: %v", err)
+	}
+
+	r.PreSampler.SetRate(rate)
+	r.PreSampler.SetError(err)
+
+	stats := r.PreSampler.Stats()
+
+	info.UpdatePreSampler(*stats)
+	info.UpdateWatchdogInfo(wi)
+
+	metrics.Gauge("datadog.trace_agent.heap_alloc", float64(wi.Mem.Alloc), nil, 1)
+	metrics.Gauge("datadog.trace_agent.cpu_percent", wi.CPU.UserAvg*100, nil, 1)
+	metrics.Gauge("datadog.trace_agent.presampler_rate", stats.Rate, nil, 1)
```

Comment:
Was thinking it would be useful to send the current waitgroup size here but the waitgroup state is private so we don't have access to it. 

Since we have rates for incoming payloads and rates for the new metric you added for how fast we're writing into the channel that should be enough information to detect when the agent is falling behind on sending traces out. Plus there will be growing memory use and then the payload refusal metrics, so looks like we're well covered on instrumentation.